build/deps/update-deps.py

```diff
@@ -347,6 +347,24 @@ def gen_git_clone(repo):
     )
 
 
+ResolvedVersion = str | None
+
+
+def get_bcr_version(name: str) -> ResolvedVersion:
```

Comment:
Let's drop this from this PR, I think it's pretty close to shipping. Feel free to add it to another PR if it ends up being useful there – I think for now we have the same high-quality auto-update mechanism as before, if we want to use BCR more at a later time we certainly can.